common/kernel/timing.cc

```diff
@@ -721,13 +737,41 @@ void TimingAnalyser::compute_criticality()
     }
 }
 
-std::vector<CellPortKey> TimingAnalyser::get_failing_eps(domain_id_t domain_pair, int count)
+void TimingAnalyser::build_detailed_net_timing_report()
 {
-    std::vector<CellPortKey> failing_eps;
+    auto &net_timings = result.detailed_net_timings;
+
+    for (domain_id_t dom_id = 0; dom_id < domain_id_t(domains.size()); ++dom_id) {
+        auto &dom = domains.at(dom_id);
+        for (auto &ep : dom.endpoints) {
+            auto &pd = ports.at(ep.first);
+            const NetInfo *net = port_info(ep.first).net;
+
+            for (auto &arr : pd.arrival) {
+                auto &launch = domains.at(arr.first).key;
+                for (auto &req : pd.required) {
+                    auto &capture = domains.at(req.first).key;
+
+                    NetSinkTiming sink_timing;
+                    sink_timing.clock_pair = ClockPair{.start = ClockEvent{.clock = launch.clock, .edge = launch.edge},
```

Comment:
I believe this officially requires a C++ version increase to C++20 to support - I'm not entirely opposed to that but I'd prefer it not just creeping in like this.